### DISCUSSION TITLE
OAuth 2 server callback without code parameter

### DISCUSSION THREAD

**Discussion by xu4wang** [directus/directus#11879] [Archive]
I'm trying to config directus using [dingtalk](https://www.dingtalk.com/en) as OAuth 2 server for log in.

The Auth request was sent to Dingtalk without any issue,
```
https://login.dingtalk.com/oauth2/challenge.htm?client_id=dingbqzziyynpbmpgtxt&scope=email&response_type=code&redirect_uri=https%3A%2F%2Fe52d-240e-47c-3028-d4fc-644c-c940-9a9c-8bf0.ngrok.io%2Fauth%2Flogin%2Fdingtalk%2Fcallback&access_type=offline&code_challenge=R_Mypgw2inmzvb94Dr3IVvpjIrqYcZVJ48PYmnbdsI0&code_challenge_method=S256&state=R_Mypgw2inmzvb94Dr3IVvpjIrqYcZVJ48PYmnbdsI0

```
The problem is after authorization, the dingtalk server redirected to directus as below:

```
https://e52d-240e-47c-3028-d4fc-644c-c940-9a9c-8bf0.ngrok.io/auth/login/dingtalk/callback?authCode=f8e79cb694d536e5b5cfea8f3dc1885e&state=R_Mypgw2inmzvb94Dr3IVvpjIrqYcZVJ48PYmnbdsI0
```

As you can see, there is no code=... but authCode=...
Is there any configuration I can do to workaround this? 

For now, I was redirected to /admin/login?reason=INVALID_USER

BR,Austin

**Comment by xu4wang** [comment]
I made a quick(and dirty) fix in Oauth2 driver.

```javascript
    try {
        res.clearCookie(`oauth2.${providerName}`);

        if ( req.query.authCode) {
            req.query.code = req.query.authCode
        }

        if (!req.query.code || !req.query.state) {
            logger.warn(`[OAuth2] Couldn't extract OAuth2 code or state from query: ${JSON.stringify(req.query)}`);
        }

        authResponse = await authenticationService.login(providerName, {
            code: req.query.code,
            codeVerifier: verifier,
            state: req.query.state,
        });
    } catch (error: any) {
        ...
```

to initialize code if there is req.query.authCode.

Seems partially solve the issue. still not fully working, since the driver cannot get userID from the code. I made some summary on the integration in #11881

**Comment by rijkvanzanten** [comment]
Lets keep this discussion going on https://github.com/directus/directus/discussions/11881 👍🏻
